mellowmax_softmax/function/boltzmax.py

```diff
@@ -1,13 +1,21 @@
-from plum import dispatch
-from numpy import ndarray
+from typing import Union
+
+import torch
+import numpy as np
 from torch import Tensor
+from numpy import ndarray
+
 
+def boltzmax(x: Union[ndarray, Tensor], beta=1) -> Union[ndarray, Tensor]:
+    if isinstance(x, ndarray):
+        x_exp = np.exp(beta * x)
 
-@dispatch
-def boltzmax(x: Tensor) -> Tensor:
-    raise NotImplementedError
+        s = np.inner(x, x_exp)
+        s /= np.sum(x_exp)
+    elif isinstance(x, Tensor):
+        x_exp = torch.exp(beta * x)
 
+        s = torch.inner(x, x_exp)
+        s /= x_exp.sum()    
 
```

Comment:
Raise a `TypeError` exception if `x` is neither a `np.ndarray` nor a `torch.Tensor`.